router/event.go

```diff
@@ -65,22 +65,25 @@ func (router *Router) eventFromRequest(r *http.Request) (*eventpkg.Event, string
 	}
 
 	event := eventpkg.New(eventType, mime, body)
+	if customevent, err := router.parseCustomEventAsCloudEvent(eventType, event.ContentType, body); err == nil {
```

Comment:
in general, this function should be called only for custom events. `if` statement should be here not inside this function.

Reply:
That increases the cyclomatic complexity to 14